internal/service/networkmanager/vpn_attachment.go

```diff
@@ -0,0 +1,323 @@
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"attachment_policy_rule_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"attachment_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"core_network_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"core_network_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"edge_location": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_account_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"segment_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"vpn_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+		},
+	}
+}
+
+func resourceVpnAttachmentCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	coreNetworkID := d.Get("core_network_id").(string)
+	vpnARN := d.Get("vpn_arn").(string)
+	input := &networkmanager.CreateSiteToSiteVpnAttachmentInput{
+		CoreNetworkId:    aws.String(coreNetworkID),
+		VpnConnectionArn: aws.String(vpnARN),
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	log.Printf("[DEBUG] Creating Network Manager VPN Attachment: %s", input)
+	output, err := conn.CreateSiteToSiteVpnAttachmentWithContext(ctx, input)
+
+	if err != nil {
+		return diag.Errorf("creating Network Manager VPN (%s) Attachment (%s): %s", vpnARN, coreNetworkID, err)
+	}
+
+	d.SetId(aws.StringValue(output.SiteToSiteVpnAttachment.Attachment.AttachmentId))
+
+	if _, err := waitVpnAttachmentCreated(ctx, conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return diag.Errorf("waiting for Network Manager VPN Attachment (%s) create: %s", d.Id(), err)
+	}
+
+	return resourceVpnAttachmentRead(ctx, d, meta)
+}
+
+func resourceVpnAttachmentRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).NetworkManagerConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	vpnAttachment, err := FindVpnAttachmentByID(ctx, conn, d.Id())
+
+	if !d.IsNewResource() && tfresource.NotFound(err) {
+		log.Printf("[WARN] Network Manager VPN Attachment %s not found, removing from state", d.Id())
```

Comment:
We have a standard format for these log messages
```suggestion
		log.Printf("[WARN] Network Manager VPN Attachment (%s) not found, removing from state", d.Id())
```